src/services/pacs_integration.tsx

```diff
@@ -0,0 +1,111 @@
+import { DcmImage, PACSMainResponse } from "../context/reducers/dicomImagesReducer";
+import axios from 'axios';
+
+declare var process: {
+    env: {
+        REACT_APP_CHRIS_UI_PFDCM_URL: string,
+    }
+};
+
+class PACSIntegration {
+
+    private static basePACSFilePath = 'SERVICES/PACS/covidnet/';
+
+    static parseResponse(rawResponse: string) {
+        let responseHeader = rawResponse.split('\r');
+
+        // the 4th element of responseHeader contains the stringified JSON body return payload
+        const responseBody = responseHeader[4];
+        const responseJSON = JSON.parse(responseBody);
+        return responseJSON;
+    }
+
+    /**
+     * Performs a PACS query on the given PatientID and returns an array of DcmImages containing metadata 
+     * of the DICOM files associated with that patient
+     * @param {string | undefined} patientID 
+     */
+    static async queryPatientFiles(patientID: string | undefined): Promise<DcmImage[]> {
+        if (!patientID) return [];
+        
+        let patientData: DcmImage[] = [];
+        try {
+            const rawResponse = await axios.post(process.env.REACT_APP_CHRIS_UI_PFDCM_URL, JSON.stringify({
+                action:"PACSinteract",
+                meta:{
+                    do:"query",
+                    on:{PatientID: patientID},
+                    PACS:"orthanc"
+                }
+            }), {headers: {'Content-Type': 'text/plain'}})
+            const parsedResponse = this.parseResponse(rawResponse.data);
+            const data: PACSMainResponse[] = parsedResponse.query.data;
+            data.forEach(study => {
+                study.series.forEach(series => {
+                    patientData.push({
+                        id: series.uid.value,
+                        creation_date: series.StudyDate.value,
+                        fname: this.basePACSFilePath + series.SeriesInstanceUID.value,
+                        PatientID: series.PatientID.value,
+                        PatientName: series.PatientName.value,
+                        PatientBirthDate: series.PatientBirthDate.value,
+                        PatientAge: series.PatientAge.value,
+                        PatientSex: series.PatientSex.value,
+                        StudyInstanceUID: series.StudyInstanceUID.value,
+                        StudyDescription: series.StudyDescription.value,
+                        SeriesInstanceUID: series.SeriesInstanceUID.value,
+                        SeriesDescription: series.SeriesDescription.value,
+                        Modality: series.Modality.value,
```

Comment:
Double checking, is it safe to assume all of those values are **always** available in the response or could there be a scenario when some missing? If some may miss, use optional chaining here as well as where you defined the interface.

I guess we won't know for sure until PACS is actually fully integrated so I'm okay with leaving the code as it is but make a note in Jira so we return to this topic when PACS integration is done.

Reply:
These values come from `pfdcm`. From my testing, whenever a value is missing in a DICOM file, `pfdcm` will still return an object with all these properties, populating any missing ones with `no value provided"`.

Reply:
Sounds good